### PR TITLE
DTSPO-2880 - adding required kv secrets for prod

### PR DESCRIPTION
**Pull request by endakelly**

### CODE REVIEW

File: components/jenkins/keyvault.tf [hunk 1]
```diff
@@ -47,3 +47,10 @@ resource "azurerm_key_vault_secret" "subscription_id" {
   value        = data.azurerm_client_config.current.subscription_id
   key_vault_id = azurerm_key_vault.jenkinskv.id
 }
+
+resource "azurerm_key_vault_secret" "env_subscription_id" {
+  count        = var.env == "ptl" ? 1 : 0
```
Comment (timja) [inline components/jenkins/keyvault.tf:52]: this should be done in the other component for all environments just prefixing it with "${var.env}-subscription-id"

**Review by timja — CHANGES_REQUESTED** [review]
see comment

File: environments/ptl/ptl.tfvars [hunk 2]
```diff
@@ -1 +1,3 @@
 private_dns_subscription_id = "1baf5470-1c3e-40d3-a6f7-74bfbce4b348"
+
+prod_subscription_id = "5ca62022-6aa2-4cee-aaa7-e7536c8d566c"
```
Comment (timja) [inline environments/ptl/ptl.tfvars:3]:
```suggestion
```

File: components/subscription-base-infra/variables.tf [hunk 3]
```diff
@@ -22,3 +22,5 @@ variable "jenkins_identity_object_id" {
   description = "Objectid for jenkins managed identity"
   default     = "7ef3b6ce-3974-41ab-8512-c3ef4bb8ae01"
 }
+
+variable "prod_subscription_id" {}
```
Comment (timja) [inline components/subscription-base-infra/variables.tf:26]:
```suggestion
```

File: components/subscription-base-infra/main.tf [hunk 4]
```diff
@@ -36,3 +36,10 @@ resource "azurerm_storage_container" "container" {
   storage_account_name  = azurerm_storage_account.storage_account.name
   container_access_type = "private"
 }
+
+resource "azurerm_key_vault_secret" "env_subscription_id" {
+  count        = var.env == "ptl" ? 1 : 0
+  name         = "${var.env}-subscription-id"
+  value        = var.prod_subscription_id
+  key_vault_id = data.azurerm_key_vault.jenkinskv.id
+}
```
Comment (timja) [inline components/subscription-base-infra/main.tf:45]:
```suggestion
}

```

File: components/subscription-base-infra/main.tf [hunk 5]
```diff
@@ -36,3 +36,10 @@ resource "azurerm_storage_container" "container" {
   storage_account_name  = azurerm_storage_account.storage_account.name
   container_access_type = "private"
 }
+
+resource "azurerm_key_vault_secret" "env_subscription_id" {
+  count        = var.env == "ptl" ? 1 : 0
+  name         = "${var.env}-subscription-id"
+  value        = var.prod_subscription_id
```
Comment (timja) [inline components/subscription-base-infra/main.tf:43]: this should come from the subscription you're current running in using a data source

File: components/subscription-base-infra/data.tf [hunk 6]
```diff
@@ -0,0 +1,4 @@
+data "azurerm_key_vault" "jenkinskv" {
+  name                = "ptl"
```
Comment (timja) [inline components/subscription-base-infra/data.tf:2]: these values are different in ptl vs ptlsbox

**Review by timja — CHANGES_REQUESTED** [review]
I think we could probably simplify this by moving back to the jenkins component and parsing some values out of this:
https://github.com/hmcts/sds-jenkins-infrastructure/blob/master/components/jenkins/locals.tf#L3-L15

**Comment by endakelly** [comment]
> I think we could probably simplify this by moving back to the jenkins component and parsing some values out of this:
> https://github.com/hmcts/sds-jenkins-infrastructure/blob/master/components/jenkins/locals.tf#L3-L15

Looking into this now

**Review by mokainos — APPROVED** [review]

**Review by timja — APPROVED** [review]
looks fine although might fail on master as i think some of these have been created already

**Pull request merged**